pkg/utils/sortedstrings_test.go

```diff
@@ -81,3 +81,41 @@ func TestSortedStringsDelete(t *testing.T) {
 	tests.Assert(t, !SortedStringHas(s, "zz"))
 
 }
+
+func TestSortedMapKeys1(t *testing.T) {
+	abc := map[string]bool{
+		"foo": true,
+		"bar": true,
+		"baz": true,
+	}
+	k := SortedMapKeys(abc)
+	tests.Assert(t, len(k) == 3, "expected len(k) == 3, got:", len(k))
+	tests.Assert(t, k[0] == "bar", `expected k[0] == "bar", got:`, k[0])
+	tests.Assert(t, k[1] == "baz", `expected k[1] == "baz", got:`, k[1])
+	tests.Assert(t, k[2] == "foo", `expected k[1] == "foo", got:`, k[2])
+}
+
+func TestSortedMapKeys2(t *testing.T) {
+	abc := map[string]bool{
+		"11 misfits":       false,
+		" this is a test ": true,
+		"flupmpf":          true,
+		"":                 false,
+	}
+	k := SortedMapKeys(abc)
```

Comment:
NACK on this test. I'd like to suggest a more correct way.

1) I think we need less of this tests.Assert custom thing, and more of "the standard way". Here's an example:

Instead of coding each specific test, thus believing we need many asserts to test, instead code the expected results in a big map or array and test that.

Eg:

```
	values := map[string]string{
		""          : "",
		"sOmEsTrInG": "SOMESTRING",
		"uppercase" : "UPPERCASE",
		// add more individual cases here...
	}

	for v, exp := range values { // run all the tests
		s := strings.ToUpper(v)
		if s != exp {
			t.Errorf("string `%s` did not match expected: `%s`", s, exp)
		}

		// NOTE: you could alternatively store some len value, and the values
		// shown in the map could be a big struct with expected things to test
		if len(s) != len(exp) {
			t.Errorf("string length `%d` did not match expected: `%d`", len(s), len(exp))
		}
	}
```

Reply:
It is a little hard to understand what the above comment and suggestion really has to do with the tests that John wrote, because your example does/tests something else, and the real argument is obfuscated by the rant about `tests.Assert`, which is in fact entirely unrelated. See how much nicer your test example would read if instead of the three line boilerplate code blocks you'd have 1 line of `tests.assert(...)` instead. ;-)

And the point you're actually making applies just the same irrespective of whether you use
```
tests.Assert(t, condition, message)
```
or
```
if !condition {
    t.Errorf(message)
    // optionally return here...
}
```
:-)

Reply:
@obnoxxx Red Hat pays me for my experience and my advice. So far you seem to not be interested in benefiting from either! You're entirely too opinionated about how this stuff should be done, for someone with far less golang experience. I'm fed up of wasting time discussing constantly instead of merging productive code.